### PR TITLE
Finish porting allmydata.web to Python 3

### PR DESCRIPTION
**Pull request by itamarst**
Fixes https://tahoe-lafs.org/trac/tahoe-lafs/ticket/3624#ticket

Relies on #988, so review that first.

### CODE REVIEW

File: src/allmydata/storage_client.py [hunk 1]
```diff
@@ -844,10 +843,10 @@ def get_available_space(self):
         version = self.get_version()
         if version is None:
             return None
-        protocol_v1_version = version.get('http://allmydata.org/tahoe/protocols/storage/v1', UnicodeKeyDict())
-        available_space = protocol_v1_version.get('available-space')
+        protocol_v1_version = version.get(b'http://allmydata.org/tahoe/protocols/storage/v1', BytesKeyDict())
```
Comment (meejah) [inline src/allmydata/storage_client.py:846]: Didn't this start out as JSON? Not sure I understand why the keys should be bytes...

File: tox.ini [hunk 2]
```diff
@@ -44,7 +44,7 @@ deps =
 # more useful results.
 usedevelop = False
 # We use extras=test to get things like "mock" that are required for our unit
-# tests.
+# tests.8
```
Comment (meejah) [inline tox.ini:47]: typo?

**Review by meejah — APPROVED** [review]
overall looks fine to me.
some inline comments / fixups / questions.

File: src/allmydata/storage_client.py (hunk 1 above)
Comment (itamarst) [reply src/allmydata/storage_client.py:846]: This comes in via remote get_version() Foolscap call, where default strings were bytes, so for wire compatibility it continues to be bytes.

**Pull request merged**